pkg/build/generator/generator_test.go

```diff
@@ -105,7 +105,7 @@ func TestInstantiateRetry(t *testing.T) {
 
 func TestInstantiateDeletingError(t *testing.T) {
 	source := mocks.MockSource()
-	generator := BuildGenerator{Client: TestingClient{
+	generator := BuildGenerator{Client: &TestingClient{
```

Comment:
why isn't this test using mockBuildGenerator() ?

Reply:
because it (and the others) weren't previously using mockBuildGenerator(), and my opinion was that letting them sit was a better bet than changing them for the sake of it.  Do you want them changed?

Reply:
I have mixed feelings, but if we're going to clean this code up, i prefer we get consistent about how we create/consume the mockGenerator.  Otherwise someone may discover this pattern and copy it elsewhere.

That said, my question was primarily "is this test doing something w/ the Generator that precludes using the mockGenerator because of the way it needs it to be created"? (e.g. mockGenerator sets up secrets also.  wasn't sure if this test could tolerate those secrets being setup)